ExampleApp/ExampleApp/Screens/ScanResults/View/UITableViewCell/ScanResultsTableViewCell.swift

```diff
@@ -0,0 +1,88 @@
+import RxBluetoothKit
+import UIKit
+
+final class ScanResultTableViewCell: UITableViewCell {
+
+    private let peripheralNameLabel: UILabel = UILabel(style: Stylesheet.Commons.titleLabel)
+
+    private let advertisementDataLabel: UILabel = UILabel(style: Stylesheet.Commons.descriptionLabel)
+
+    private let rssiLabel: UILabel = UILabel(style: Stylesheet.Commons.descriptionLabel)
+
+    private let bluetoothImageView: UIImageView = UIImageView(image: Constant.ImageRepo.bluetooth)
+
+    private let connectButton: UIButton = UIButton(style: Stylesheet.Commons.blackButton)
+
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: .default, reuseIdentifier: reuseIdentifier)
+        applyStyles()
+        setConstraints()
+        backgroundColor = .white
+        selectionStyle = .none
+        connectButton.setTitle("Connect", for: .normal)
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        peripheralNameLabel.text = nil
+        advertisementDataLabel.text = nil
+        rssiLabel.text = nil
+    }
+
+    func setConnectTarget(_ target: Any, action: Selector, for events: UIControlEvents) {
+        if connectButton.allTargets.isEmpty {
+            connectButton.addTarget(target, action: action, for: events)
```

Comment:
Not really sure what it'd take to change it, but I must admit I'm pretty surprised to see a targed-action based communication in a RxSwift-heavy app :)

Reply:
Yup, i understand that is so there is no need for handling disposeBag in VC when cell is reusing :)
If that is the case maybe consider providing some `connectCellAction` in configuration of cell so disposeBag would stay in cell class and you would provide the action from VC inside closure?